core/src/main/java/org/keycloak/sdjwt/SdJwtVerificationContext.java

```diff
@@ -142,12 +143,15 @@ public void verifyPresentation(
         }
 
         // Upon receiving a Presentation, in addition to the checks in {@link #verifyIssuance}...
-        verifyIssuance(issuerSignedJwtVerificationOpts);
+        JsonNode disclosedPayload = verifyIssuance(issuerVerifyingKeys, issuerSignedJwtVerificationOpts);
 
         // Validate Key Binding JWT if required
         if (keyBindingJwtVerificationOpts.isKeyBindingRequired()) {
             validateKeyBindingJwt(keyBindingJwtVerificationOpts);
         }
+
+        // Return Issuer-signed JWT's disclosed payload
+        return disclosedPayload;
```

Comment:
Whats the reason for returning the disclosed payload as result of the verification method? Its not what I would necessarily expect from a verification method. My expectation from the method signature would either be no content or the complete verified object. 
Beside that, since its a presentation that gets verified, will it be the payload of all contained credentials and how are the payloads seperated then?

Reply:
> Whats the reason for returning the disclosed payload as result of the verification method? Its not what I would necessarily expect from a verification method. My expectation from the method signature would either be no content or the complete verified object.

The disclosed payload is needed to run some checks on the presence or absence of some expected claims. I agree that returning it from here is not required to run the logic. I updated the code with an alternative means, dismissing the return.

> Beside that, since its a presentation that gets verified, will it be the payload of all contained credentials and how are the payloads seperated then?

This class is about verifying a single SD-JWT VP; and from the best of my knowledge, it cannot embed multiple credentials, hence a single disclosed payload.